File: backend/internal/model/series.go
package model

// Series represents a TV show
type Series struct {
	ID             int64   `json:"id"`
	LibraryID      int64   `json:"library_id"`
	Title          string  `json:"title"`
	SortTitle      string  `json:"sort_title"`
	TmdbID         *int64  `json:"tmdb_id,omitempty"`
	ImdbID         *string `json:"imdb_id,omitempty"`
	TvdbID         *int64  `json:"tvdb_id,omitempty"`
	Overview       string  `json:"overview"`
	Status         string  `json:"status"`         // "Returning Series" | "Ended" | "Canceled"
	Network        string  `json:"network"`        // "CBS", "Netflix", etc.
	FirstAirDate   string  `json:"first_air_date"` // YYYY-MM-DD
	PosterPath     string  `json:"poster_path"`
	BackdropPath   string  `json:"backdrop_path"`
	LogoPath       string  `json:"logo_path"`
	ThumbPath      string  `json:"thumb_path"`
	MetadataLocked bool    `json:"metadata_locked"`
	CreatedAt      string  `json:"created_at"`
	UpdatedAt      string  `json:"updated_at"`
}

// SeriesMetadataEditRequest represents a partial metadata edit for series.
type SeriesMetadataEditRequest struct {
	Title          *string       `json:"title"`
	SortTitle      *string       `json:"sort_title"`
	Overview       *string       `json:"overview"`
	Status         *string       `json:"status"`
	Network        *string       `json:"network"`
	FirstAirDate   *string       `json:"first_air_date"`
	Genres         []string      `json:"genres"`
	Credits        []CreditInput `json:"credits"`
	SaveNFO        bool          `json:"save_nfo"`
	MetadataLocked *bool         `json:"metadata_locked"`
}

// Season represents a season of a series
type Season struct {
	ID           int64  `json:"id"`
	SeriesID     int64  `json:"series_id"`
	SeasonNumber int    `json:"season_number"`
	Title        string `json:"title"`
	Overview     string `json:"overview"`
	PosterPath   string `json:"poster_path"`
	EpisodeCount int    `json:"episode_count"`
	CreatedAt    string `json:"created_at"`
}

// Episode represents a single episode linking to a media item
type Episode struct {
	ID            int64  `json:"id"`
	SeriesID      int64  `json:"series_id"`
	SeasonID      int64  `json:"season_id"`
	MediaID       int64  `json:"media_id"`
	EpisodeNumber int    `json:"episode_number"`
	Title         string `json:"title"`
	Overview      string `json:"overview"`
	StillPath     string `json:"still_path"`
	AirDate       string `json:"air_date"` // YYYY-MM-DD
	CreatedAt     string `json:"created_at"`
}

// EpisodeWithMedia combines episode with its media and media file
type EpisodeWithMedia struct {
	Episode     Episode    `json:"episode"`
	Media       Media      `json:"media"`
	PrimaryFile *MediaFile `json:"primary_file,omitempty"`
}

// SeriesWithSeasons combines series with its seasons
type SeriesWithSeasons struct {
	Series  Series   `json:"series"`
	Seasons []Season `json:"seasons"`
}

File: packages/shared/types/series.ts
// Series, Season, Episode types

import type { MediaFile, CreditInput } from './media'

export interface Series {
  id: number
  library_id: number
  title: string
  sort_title: string
  tmdb_id?: number
  imdb_id?: string
  tvdb_id?: number
  overview: string
  status: string // "Returning Series" | "Ended" | "Canceled"
  network: string
  first_air_date: string
  poster_path: string
  backdrop_path: string
  logo_path: string
  thumb_path: string
  metadata_locked: boolean
  created_at: string
  updated_at: string
}

export interface SeriesListItem {
  id: number
  library_id: number
  title: string
  sort_title: string
  tmdb_id?: number
  imdb_id?: string
  tvdb_id?: number
  overview: string
  status: string
  network: string
  first_air_date: string
  poster_path: string
  backdrop_path: string
  logo_path: string
  thumb_path: string
  metadata_locked: boolean
  created_at: string
  updated_at: string
  // Extra fields beyond Series
  genres: string[]
  season_count?: number
  episode_count?: number
}

export interface SeriesWithSeasons {
  series: Series
  seasons: Season[]
}

export interface SeriesListParams {
  library_id?: number
  search?: string
  genre?: string
  year?: string
  sort?: 'newest' | 'oldest' | 'rating' | 'title'
  limit?: number
  offset?: number
}

export interface SeriesMetadataEditRequest {
  title?: string
  sort_title?: string
  overview?: string
  status?: string
  network?: string
  first_air_date?: string
  genres?: string[]
  credits?: CreditInput[]
  save_nfo?: boolean
  metadata_locked?: boolean
}

export interface EpisodeMetadataEditRequest {
  title?: string
  overview?: string
  air_date?: string
  episode_number?: number
  metadata_locked?: boolean
}

export interface Season {
  id: number
  series_id: number
  season_number: number
  title: string
  overview: string
  poster_path: string
  air_date?: string
  episode_count: number
  created_at: string
  updated_at: string
}

export interface Episode {
  id: number
  series_id: number
  season_id: number
  media_id: number
  episode_number: number
  title: string
  overview: string
  still_path?: string
  air_date?: string
  duration?: number
  media_files?: MediaFile[]
  created_at: string
  updated_at: string
}
